media-library-client/src/app/private-container/pages/home/video-gallery/video-tiles/video-tiles.tsx

```diff
@@ -39,33 +40,150 @@ export default function VideoTiles() {
 				route: RouteVideos(videoGallery?.id),
 			},
 		]);
+		console.log('Breadcrumb setup');
+		console.log('Fetching videos');
 	}, [videoGallery]);
 
-	useEffect(() => {
-		if (videoGalleryGetAllQuery.isSuccess && videoGalleryGetAllQuery.data) {
-			setVideoGallery(videoGalleryGetAllQuery.data);
-		}
-	}, [videoGalleryGetAllQuery.status, videoGalleryGetAllQuery.data]);
+	const { data: videoGalleryData, isSuccess: isVideoGallerySuccess } = useQuery({
+		...HttpHelper.videoGallery.GetByID(videoGalleryID),
+		enabled: true,
+	});
+
+	const { data: videosData } = useQuery({
+		...HttpHelper.video.GetByPage(videoGalleryID, pageSetup.page, pageSetup.pageSize),
+		enabled: !!videoGallery,
+	});
+
+	const onClickActiveVideo = (video: Video) => {
+		setVideoStream(HttpHelper.videoStream.GetVideoStream(video.id));
+	};
 
 	useEffect(() => {
-		if (videoGetByGalleryQuery.isSuccess && videoGetByGalleryQuery.data) {
-			setVideos(videoGetByGalleryQuery.data);
+		console.log('trying to set Video Gallery');
+		if (isVideoGallerySuccess && videoGalleryData) {
+			console.log(`set Video Gallery as ${JSON.stringify(videoGalleryData)}`);
+			setVideoGallery(videoGalleryData);
 		}
-	}, [videoGetByGalleryQuery.status, videoGetByGalleryQuery.data]);
+	}, [isVideoGallerySuccess, videoGalleryData]);
 
-	const renderTiles = useMemo(() => {
-		const components: React.JSX.Element[] = [];
-		if (videos) {
-			for (const video of videos) {
-				components.push(<Box key={video.id}>{video.name}</Box>);
-			}
+	const renderPlayer = () => {
+		if (videoStream) {
+			console.log(`videoStream`, videoStream);
+			return <VideoPlayer URL={videoStream}></VideoPlayer>;
+		} else {
+			return <Typography>No video has been selected.</Typography>;
 		}
-		return components;
-	}, [videos]);
+	};
+
+	const renderVideos = useMemo(() => {
+		console.log(`viodeos : ${JSON.stringify(videosData)}`);
+		if (!videosData) return null;
+
+		return videosData.map(video => (
+			<Grid key={videoGallery?.id} container sx={{ border: 1, borderRadius: 2, p: 2, mb: 2 }}>
+				<Grid item xs={6}>
+				<IconButtonWrapper id={'edit'} icon={IconSelector.Edit} sx={{ margin: 1, display: isEdit ? 'none' : 'inline-flex' }} onClick={() => setIsEdit(true)} />
+				<IconButtonWrapper id={'save'} icon={IconSelector.Save} sx={{ margin: 1, display: isEdit ? 'inline-flex' : 'none' }} onClick={() => {
+					if (videoGallery) {
+						HttpHelper.videoGallery.Save(videoGallery).then(res => alert(JSON.stringify(res))).catch(err => alert(JSON.stringify(err)));

```

Comment:
Use the "httpOptions" param of the "HttpHelper.videoGallery.Save" method instead of "then" and "catch"